spring-cloud-gcp-docs/src/main/asciidoc/datastore.adoc

```diff
@@ -309,6 +311,47 @@ Since Cloud Datastore entity keys can have multiple parents, it is possible that
 Because entity keys are immutable in Cloud Datastore, to change the key of a child you must delete the existing one and re-save it with the new key.
 
 
+==== Explicit Key-Column Reference Relationships
+
+General relationships can be stored using the `@Reference` annotation.
+
+[source, java]
+----
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Reference;
+import org.springframework.data.annotation.Id;
+
+@Entity
+public class ShoppingOrder {
+  @Id
+  long id;
+
+  @Reference
+  List<Item> items;
+
+  @Reference
+  Item specialSingleItem;
+}
+
+@Entity
+public class Item {
+  @Id
+  Key purchasedItemKey;
+
+  String name;
+
+  Timestamp timeAddedToOrder;
+}
+----
+
+Reference properties can either be singular or collection-like.
+These properties correspond to actual columns in the entity and Cloud Datastore Kind that hold the key values of the referenced entities.
+The referenced entities are full-fledged entities of other Kinds.
+
+Similar to the `@Descendants` relationships, reading or writing an entity will recursively read or write all of the referenced entities at all levels.
+If referenced entities have `null` ID values, then they will be saved as new entities and will have ID values allocated by Cloud Datastore.
+There are no requirements for relationships between the key of an entity and the keys that entity holds as references.
```

Comment:
mention that the order of the items in the referenced collection can be different on load

Reply:
Yes good point. wil do